app/src/test/java/org/dhis2/usescases/settings/SettingsRepositoryTest.kt

```diff
@@ -0,0 +1,329 @@
+package org.dhis2.usescases.settings
+
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.Single
+import org.dhis2.Bindings.toSeconds
+import org.dhis2.data.prefs.Preference.Companion.DEFAULT_NUMBER_RV
+import org.dhis2.data.prefs.Preference.Companion.EVENT_MAX
+import org.dhis2.data.prefs.Preference.Companion.EVENT_MAX_DEFAULT
+import org.dhis2.data.prefs.Preference.Companion.LIMIT_BY_ORG_UNIT
+import org.dhis2.data.prefs.Preference.Companion.LIMIT_BY_PROGRAM
+import org.dhis2.data.prefs.Preference.Companion.NUMBER_RV
+import org.dhis2.data.prefs.Preference.Companion.TEI_MAX
+import org.dhis2.data.prefs.Preference.Companion.TEI_MAX_DEFAULT
+import org.dhis2.data.prefs.Preference.Companion.TIME_15M
+import org.dhis2.data.prefs.Preference.Companion.TIME_DAILY
+import org.dhis2.data.prefs.Preference.Companion.TIME_DATA
+import org.dhis2.data.prefs.Preference.Companion.TIME_META
+import org.dhis2.data.prefs.Preference.Companion.TIME_WEEKLY
+import org.dhis2.data.prefs.PreferenceProvider
+import org.dhis2.utils.Constants
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.common.State
+import org.hisp.dhis.android.core.settings.DataSyncPeriod
+import org.hisp.dhis.android.core.settings.GeneralSettings
+import org.hisp.dhis.android.core.settings.LimitScope
+import org.hisp.dhis.android.core.settings.MetadataSyncPeriod
+import org.hisp.dhis.android.core.settings.ProgramSetting
+import org.hisp.dhis.android.core.settings.ProgramSettings
+import org.hisp.dhis.android.core.sms.domain.interactor.ConfigCase
+import org.hisp.dhis.android.core.sms.domain.repository.WebApiRepository
+import org.hisp.dhis.android.core.sms.domain.repository.internal.LocalDbRepository
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito
+
+class SettingsRepositoryTest {
+
```

Comment:
Does this class pass the ktlint Check?

Reply:
:open_mouth:

Reply:
Yes it did. But other failed so I fixed them